packages/main-api/src/types/membership.rs

```diff
@@ -14,8 +14,150 @@ use bdk::prelude::*;
 pub enum Membership {
     #[default]
     Free = 1,
-    Paid1 = 2,
-    Paid2 = 3,
-    Paid3 = 4,
+    Pro = 2,
+    Max = 3,
+    VIP = 4,
+    Enterprise = 5,
     Admin = 99,
 }
+
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, Default)]
+pub struct MembershipInfo {
+    pub membership_type: Membership,
+    pub subscription_start: i64, // timestamp in microseconds
+    pub subscription_end: i64,   // timestamp in microseconds
+    /// HashMap where key is booster type (0=no boost, 2=2x, 10=10x, 100=100x, 1000=1000x) and value is remaining quota for the month
+    pub space_capabilities: HashMap<u32, i32>,
+}
+
+impl MembershipInfo {
+    pub fn new_free() -> Self {
```

Comment:
Use Builder or factory liked pattern.

For builder-like pattern,

Membership::new().with_pro();

Or
Membership::new().with_enterprise(custom_boosters);
@blikq